### PR TITLE
Do not use virtual edges for building isochrone geometry

### PR DESCRIPTION
**Pull request by HendrikLeuschner**
### Pull Request Checklist

- [x] 1. I have [**rebased**](https://github.com/GIScience/openrouteservice/blob/master/CONTRIBUTE.md#pull-request-guidelines) the latest version of the master branch into my feature branch and all conflicts have been resolved.
- [x] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the [Unreleased] heading.
- [x] 3. I have documented my code using JDocs tags.
- [x] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [x] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [x] 6. I have created API tests for any new functionality exposed to the API.
- [x] 7. If changes/additions are made to the app.config file, I have added these to the app.config wiki page on github along with a short description of what it is for, and documented this in the Pull Request (below).
- [x] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [x] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [x] 10. For new features or changes involving building of graphs, I have tested on a larger dataset (at least Germany) and the graphs build without problems (i.e. no out-of-memory errors).
- [x] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the importer etc.), I have generated longer distance routes for the affected profiles with different options (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end points generated from the current live ORS. If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [x] 12. I have written in the Pull Request information about the changes made including their intended usage and why the change was needed.
- [x] 13. For changes touching the API documentation, i have tested that the API playground [renders correctly](https://gitlab.gistools.geog.uni-heidelberg.de/giscience/openrouteservice-infrastructure/ors-docs-api#test-new-ors-documentation).

Fixes # .

### Information about the changes
- Key functionality added: Check whether edge is virtual before processing in isochrone geometry. Virtual edges have no geometry -> fails. Was added because one of the last changes added QueryGraph usage for isochrones for fixing another bug.
- Reason for change:

### Examples and reasons for differences between live ORS routes and those generated from this pull request
-

### Required changes to app.config (if applicable)
-

### CODE REVIEW

File: openrouteservice/src/main/java/org/heigit/ors/isochrones/builders/concaveballs/ConcaveBallsIsochroneMapBuilder.java [hunk 1]
```diff
@@ -332,6 +332,7 @@ private GeometryCollection buildIsochrone(AccessibilityMap edgeMap, List<Coordin
 		GraphHopperStorage graph = searchContext.getGraphHopper().getGraphHopperStorage();
 		NodeAccess nodeAccess = graph.getNodeAccess();
 		int maxNodeId = graph.getNodes();
```
Comment (aoles) [inline openrouteservice/src/main/java/org/heigit/ors/isochrones/builders/concaveballs/ConcaveBallsIsochroneMapBuilder.java]: In node indexing is 0-based and `getNodes()` returns the number of nodes, then shouldn't this rather be `int maxNodeId = graph.getNodes() - 1;`?

**Review by aoles — CHANGES_REQUESTED** [review]

File: openrouteservice/src/main/java/org/heigit/ors/isochrones/builders/concaveballs/ConcaveBallsIsochroneMapBuilder.java (hunk 1 above)
Comment (HendrikLeuschner) [reply openrouteservice/src/main/java/org/heigit/ors/isochrones/builders/concaveballs/ConcaveBallsIsochroneMapBuilder.java]: Citing GH 
```The virtual nodes and edges have a higher int ID than graph.getNodes() or allEdges.length()``` https://github.com/graphhopper/graphhopper/blob/master/docs/core/low-level-api.md
This is at least for the edges not true. There it really is the count. The node part was already in and I did not change or check it.

Comment (aoles) [reply openrouteservice/src/main/java/org/heigit/ors/isochrones/builders/concaveballs/ConcaveBallsIsochroneMapBuilder.java]: Apparently that's not entirely true according to https://github.com/GIScience/graphhopper/blob/ors_0.13/core/src/main/java/com/graphhopper/routing/QueryGraph.java#L375-L381

Comment (HendrikLeuschner) [reply openrouteservice/src/main/java/org/heigit/ors/isochrones/builders/concaveballs/ConcaveBallsIsochroneMapBuilder.java]: Yes, as I said, for the edges it is not true (I use id > maxId -1, QueryGraph uses id >= maxId). So seems that its also not true for the nodes, if we can trust QueryGraph impl. So generally the GH explanation is wrong.

Comment (HendrikLeuschner) [reply openrouteservice/src/main/java/org/heigit/ors/isochrones/builders/concaveballs/ConcaveBallsIsochroneMapBuilder.java]: Changed it

Comment (aoles) [reply openrouteservice/src/main/java/org/heigit/ors/isochrones/builders/concaveballs/ConcaveBallsIsochroneMapBuilder.java]: I believe both `isVirtualEdge` and `isVirtualNode` methods are correct considering other bits of GH code I've seen such as https://github.com/GIScience/graphhopper/blob/ors_0.13/core/src/main/java/com/graphhopper/storage/BaseGraph.java#L69-L72 and https://github.com/GIScience/graphhopper/blob/ors_0.13/core/src/main/java/com/graphhopper/storage/BaseGraph.java#L86-89, respectively.

**Review by aoles — APPROVED** [review]
LGTM, thanks for fixing this!

**Pull request merged**